### PR TITLE
Fluent checking (fix bugs)

### PR DESCRIPTION
**Pull request by SebastianSchmidl**

### CODE REVIEW

File: distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala [hunk 1]
```diff
@@ -36,46 +37,106 @@ class SplitCandidateValidationBehavior(
...
-      val result = checkSplitCandidates(candidateId, splitCandidates, attributes, errors)
+  private def changeToChecking(errorCompare: Double, errors: Map[CandidateSet, Double], toBeChecked: CandidateSet): Behavior[Command] = {
+    spans.begin("Split check")
+    val candidates = for {
+      a <- toBeChecked.unsorted
+      context = candidateId - a
+      errorContext <- errors.get(context)
+    } yield a -> checkSplitCandidate(errorContext, errorCompare)
+
+    val validCandidates = candidates.filter(_._2).map(_._1)
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala [hunk 2]
```diff
@@ -36,46 +37,106 @@ class SplitCandidateValidationBehavior(
...
+      case WrappedPartitionEvent(ErrorFound(`candidateId`, _)) =>
+        // ignore, should not happen
+        context.log.error("Received unexpected")
+        Behaviors.same
+
+      case WrappedPartitionEvent(ErrorFound(key, value)) =>
+        context.log.trace("Received partition error value: {}, {}", key, value)
+        spans.begin("Split check")
+        val attributeSet = candidateId diff key
+        val attribute = attributeSet.head
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/test/scala/com/github/codelionx/distod/actors/partitionMgmt/JobChainerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package com.github.codelionx.distod.actors.partitionMgmt
+
+import com.github.codelionx.distod.Settings.PartitionCompactionSettings
+import com.github.codelionx.distod.partitions.{FullPartition, StrippedPartition}
+import com.github.codelionx.distod.types.CandidateSet
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.concurrent.duration.FiniteDuration
```
Comment (CodeLionX) [inline distod/src/test/scala/com/github/codelionx/distod/actors/partitionMgmt/JobChainerSpec.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala [hunk 4]
```diff
@@ -36,46 +37,106 @@ class SplitCandidateValidationBehavior(
...
+  private def changeToChecking(errorCompare: Double, errors: Map[CandidateSet, Double], toBeChecked: CandidateSet): Behavior[Command] = {
+    spans.begin("Split check")
+    val candidates = for {
+      a <- toBeChecked.unsorted
+      context = candidateId - a
+      errorContext <- errors.get(context)
+    } yield a -> checkSplitCandidate(errorContext, errorCompare)
+
+    val validCandidates = candidates.filter(_._2).map(_._1)
+    val remainingCandidates = toBeChecked diff candidates.map(_._1)
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/test/scala/com/github/codelionx/distod/actors/partitionMgmt/JobChainerSpec.scala (hunk 3 above)
Comment (SebastianSchmidl) [inline distod/src/test/scala/com/github/codelionx/distod/actors/partitionMgmt/JobChainerSpec.scala:9]: Codacy Issue found: [Use brackets to group imports from the same package](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala [hunk 5]
```diff
@@ -36,46 +37,108 @@ class SplitCandidateValidationBehavior(
...
+  ): Behavior[Command] = {
+    spans.begin("Split check")
+    val candidates = for {
+      a <- toBeChecked.unsorted
+      context = candidateId - a
+      errorContext <- errors.get(context)
+    } yield a -> checkSplitCandidate(errorContext, errorCompare)
+
+    val validCandidates = candidates.filter(t => t._2).map(t => t._1)
+    val remainingCandidates = toBeChecked diff candidates.map(t => t._1)
```
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala:117]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

File: distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala [hunk 6]
```diff
@@ -36,46 +37,108 @@ class SplitCandidateValidationBehavior(
...
+      errorCompare: Double, errors: Map[CandidateSet, Double], toBeChecked: CandidateSet
+  ): Behavior[Command] = {
+    spans.begin("Split check")
+    val candidates = for {
+      a <- toBeChecked.unsorted
+      context = candidateId - a
+      errorContext <- errors.get(context)
+    } yield a -> checkSplitCandidate(errorContext, errorCompare)
+
+    val validCandidates = candidates.filter(t => t._2).map(t => t._1)
```
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala:116]: Codacy Issue found: [Found unnamed result or parameter](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)

**Pull request merged**